### PR TITLE
[VolumetricRendering][SofaCUDA] Move CUDA files related to VolumetricRendering

### PR DESCRIPTION
**Pull request by alxbilger**
… into a dedicated plugin

The plugin VolumetricRendering needed a bit of cleaning too.

The new folder hierarchy is the following:

- SofaCUDA
- VolumetricRendering
  - VolumetricRendering
  - VolumetricRendering.CUDA

______________________________________________________

By submitting this pull request, I acknowledge that  
**I have read, understand, and agree [SOFA Developer Certificate of Origin (DCO)](https://github.com/sofa-framework/sofa/blob/master/CONTRIBUTING.md#sofa-developer-certificate-of-origin-dco)**.
______________________________________________________

**Reviewers will merge this pull-request only if**  
- it builds with SUCCESS for all platforms on the CI.
- it does not generate new warnings.
- it does not generate new unit test failures.
- it does not generate new scene test failures.
- it does not break API compatibility.
- it is more than 1 week old (or has fast-merge label).

### CODE REVIEW

**Comment by alxbilger** [comment]
[ci-build][with-all-tests]

**Comment by bakpaul** [comment]
The usual way of adding cuda support for cpu-based components was done through the SofaCUDA plugin. Many component of this plugin are compiled only if the cpu-based plugin of those components is found. This allowed to keep all the cuda-based code in the same plugin.

What you propose here goes against this usual way of developing cuda based components. This has one important effect, it adds cuda-based codes in base plugins (as you did here, even if it is in sub project). Why doing it for this plugin and not the rest ?

I am not saying that what you propose is wrong, but that only one approach should be applied to the whole plugins either the old one or yours, to ease future developments. It may be interesting to discuss it with the other devs.

**Comment by alxbilger** [comment]
Sorry I should have given more details. The move I suggest is something that was around for quite a while now. I think it was suggested by @damienmarchal first.

The legacy conditionally compiled code, mainly in SofaCUDA but it applies everywhere, is a bad practice for the following reasons:
- It is nearly impossible for the user to know what components are included in the conditionally compiled module. It would require a complex documentation to inform the user, which is not done for SofaCUDA.
- Testing becomes more complex. It's challenging to ensure that all possible combinations of conditions are thoroughly tested, increasing the risk of undetected bugs. See https://github.com/sofa-framework/sofa/pull/4227 for example.
- Other plugins (for example SoftRobots) cannot add their own CUDA components into SofaCUDA. So it's not true that all components are in the same plugin. We also have the example of the GPU-based solvers.

What I suggest is to keep the cpu-based plugin, and create a new plugin for the gpu-based components. It would be applied to all plugins that are a conditional dependency in SofaCUDA (so not the `Sofa.Component.*` modules), not only `VolumetricRendering`. This PR is an illustration of the move of one particular plugin, but it should be generalized. The goal of the PR is also to approve the move, but also the folder architecture. If we can agree on an architecture inside the SOFA codebase, we can also apply it in external plugins (such as SoftRobots or MOR).

File: applications/plugins/VolumetricRendering/VolumetricRendering/CMakeLists.txt [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+    src/VolumetricRendering/OglTetrahedralModel.cpp
+    src/VolumetricRendering/OglVolumetricModel.cpp
+)
+
+add_library(${PROJECT_NAME} SHARED  ${HEADER_FILES} ${SOURCE_FILES})
+
+target_link_libraries(${PROJECT_NAME} Sofa.Helper Sofa.GL.Component)
+
+# Add the include DIR from OpenHaptics to target
+target_include_directories(${PROJECT_NAME} PUBLIC "$<BUILD_INTERFACE:${OPENHAPTICS_INCLUDE_DIR}>")
```
Comment (fredroy) [inline applications/plugins/VolumetricRendering/VolumetricRendering/CMakeLists.txt:25]:
```suggestion
```

**Review by fredroy — CHANGES_REQUESTED** [review]

**Comment by damienmarchal** [comment]
Drafted ideas.

    SofaCUDA
    VolumetricRendering
        VolumetricRendering
             CMakeLists.txt
             src
             examples
        VolumetricRendering.CUDA
             CMakeLists.txt
             src
             examples
    
    SofaCaribou
       CMakeLists.txt
       src
       examples
             
    SofaPython3
    Sofa.Component.IO.Mesh
       CMakeLists.txt
       src
       examples
    Sofa.Component.IO.Mesh.Python
       CMakeLists.txt (depend on: Sofa.Component.IO.Mesh, SofaPython3)
       src
       examples
         
    SofaCaribou.Python
       CMakeLists.txt (depend on: Caribou, SofaPython3)
       src
       examples
    
    Cosserat
       CMakeLists.txt
       src
       examples
   
    Cosserat.Python
       CMakeLists.txt
       src
       examples

    Cosserat
       Cosserat/
            CMakeLists.txt
            src/
            examples/
       
       Cosserat.CUDA/
            CMakeLists.txt
            src
            examples
       
       PythonBinding/
                  CMakeLists.txt
                  src
                  examples

**Pull request closed without merging**